src/udev/udev-builtin-input_id.c

```diff
@@ -240,13 +241,18 @@ static bool test_pointers(struct udev_device *dev,
                         is_touchscreen = true;
         }
 
-        if (has_rel_coordinates && has_mouse_button)
-                is_mouse = true;
+        if (has_mouse_button)
+                if (has_rel_coordinates)
+                        is_mouse = true;
+                if (!has_abs_coordinates) /* exclude vmmouse and joysticks */
```

Comment:
Did you mean to drop the `!` here? I'd certainly define a "pad" to be a device with mouse buttons and absolute coordinates, not the absence of the latter.